### DISCUSSION TITLE
Get ServerSideProps with page level HOC example.

### DISCUSSION THREAD

**Discussion by sambhav-gore** [vercel/next.js#12927] [Help]
Hi, 
I have built my solution around this example - https://github.com/zeit/next.js/tree/canary/examples/api-routes-apollo-server-and-client

that makes use of an HOC (withApollo) to add the apolloClient to the page.

now on the page, if I want to use getServerSideProps, it is not possible because the two cannot work together.

any suggestions? If I keep using getInitialProps on my pages, is it a problem?

**Comment by leerob** [comment]
> Here's a minimal example for fetching a GraphQL API: https://github.com/zeit/next.js/blob/canary/examples/cms-datocms/lib/api.js#L23-L42

> The way that Apollo does tree-rendering to allow useQuery causes quite a lot of performance overhead (and general bundle size when using Apollo is quite heavy currently). So would recommend using a lightweight solution.

https://github.com/zeit/next.js/discussions/10946

**Reply by sambhav-gore** [reply]
thanks.
